### PR TITLE
chore: release scripts for travis

### PR DESCRIPTION
**Pull request by efegurkan**
#### Reference issue: N/A

This PR adds changes to the release scripts and travis config to enable testing melody easier and more flexible at trivago.

With this changes, every push to the master should release a new version to the `@next` tag with next minor version + commit id. Also for the any branch push it should create a release to the `@canary` tag. 

This also includes a tiny peer dependency updating utility and some prompts before releasing. That should prevent unnecessary patch releases in future.

### CODE REVIEW

File: bin/ci-release.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+if [ "$TRAVIS_BRANCH" = "master" ]; then
+  # RELEASE AS @NEXT
+  yarn install && \
+    yarn lerna publish --skip-git --skip-npm --exact --force-publish=* --canary=commit --npm-tag=next --registry https://registry.npmjs.org
+  exit 0
+else
+  # CANARY RELEASE WITH COMMIT NAME
+  # @commit
+  yarn install && \
+    yarn lerna publish --skip-git --skip-npm --exact --force-publish=*  --canary=commit --registry http://localhost:4873/
```
Comment (ayusharma) [inline bin/ci-release.sh]: localhost:4873 ?

Comment (ayusharma) [reply bin/ci-release.sh]: I think it's verdaccio's URL .

Comment (efegurkan) [reply bin/ci-release.sh]: true, I forgot to change that one

**Comment by efegurkan** [comment]
This is ready for the review. 

@ayusharma please feel free to jump in and comment.

File: bin/ci-release.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    echo
+    echo "USER AUTHENTICATION FAILED"
+    echo
+    exit 1
+fi
+
+if [ "$TRAVIS_BRANCH" = "master" ]; then
+  # RELEASE AS @NEXT
+  yarn install && \
+    yarn lerna publish --skip-git --exact --force-publish=* --canary=commit --npm-tag=next --registry https://registry.npmjs.org
```
Comment (ayusharma) [inline bin/ci-release.sh]: I am not sure about `--exact`. There is nothing mentioned in docs https://lernajs.io/#command-publish

File: bin/updatePeerDependencies.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if (!peerDeps || !Object.keys(peerDeps).length) {
+        console.log(
+            pkgJson.name,
+            chalk.keyword('orange')('~ No peer dependencies. Skipping.')
+        );
+        return;
+    }
+
+    Object.keys(pkgJson.peerDependencies).forEach(dep => {
+        if (pkgNames.indexOf(dep) > -1) {
```
Comment (ayusharma) [inline bin/updatePeerDependencies.js]: Better to use `.includes`. 😄

**Comment by pago** [comment]
@efegurkan Can we get the automated tests to pass?

File: bin/ci-release.sh (hunk 2 above)
Comment (efegurkan) [reply bin/ci-release.sh]: Melody uses 2.x version of Lerna still.  https://github.com/lerna/lerna/tree/2.x#--exact

**Comment by efegurkan** [comment]
@pago It has to be just rebased onto master. I have it locally now, I don't have push rights to a branch anymore.

**Comment by pago** [comment]
```
$ bundlesize
 ERROR  Could not add github status.
        404: Not Found 
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command "yarn bundlesize" failed 3 times.
The command "travis_retry yarn bundlesize" exited with 1.
```

**Comment by ayusharma** [comment]
@pago It's a bug. May be `bundlesize` version bump can help here. https://github.com/siddharthkp/bundlesize/issues/93#issuecomment-325190802

**Comment by efegurkan** [comment]
@pago can be also bundlesize token bound to my account?  Does it fail on other PR's as well. In that case you need to renew the token.

**Comment by pago** [comment]
Almost there.
@jamesb3ll please add `trivagonpm` and me as a maintainer for `neutrino-preset-melody`

I'm not sure why but the npm website currently doesn't list `trivagonpm` as a maintainer of anything anymore. And @efegurkan seems to have been lost as well.

Once that is done, travis should pass, leaving us with the broken circleci tests on node 6.

**Pull request merged**